src/prefect/tasks/snowflake/snowflake.py

```diff
@@ -114,3 +115,117 @@ def run(self, query: str = None, data: tuple = None, autocommit: bool = None):
         except Exception as error:
             conn.close()
             raise error
+
+
+class SnowflakeQueriesFromFile(Task):
+    """
+    Task for executing queries loaded from a file against a Snowflake database.
+    Return a list containings the results of the queries.
+
+    Args:
+        - account (str): snowflake account name, see snowflake connector
+             package documentation for details
+        - user (str): user name used to authenticate
+        - password (str, optional): password used to authenticate.
+            password or private_lkey must be present
+        - private_key (bytes, optional): pem to authenticate.
+            password or private_key must be present
+        - database (str, optional): name of the default database to use
+        - schema (int, optional): name of the default schema to use
+        - role (str, optional): name of the default role to use
+        - warehouse (str, optional): name of the default warehouse to use
+        - file_path (str, optional): file path to load query from
+        - autocommit (bool, optional): set to True to autocommit, defaults to None, which
+            takes snowflake AUTOCOMMIT parameter
+        - **kwargs (dict, optional): additional keyword arguments to pass to the
+            Task constructor
+    """
+
+    def __init__(
+        self,
+        account: str,
+        user: str,
+        password: str = None,
+        private_key: bytes = None,
+        database: str = None,
+        schema: str = None,
+        role: str = None,
+        warehouse: str = None,
+        file_path: str = None,
+        autocommit: bool = None,
+        **kwargs
+    ):
+        self.account = account
+        self.user = user
+        self.password = password
+        self.database = database
+        self.schema = schema
+        self.role = role
+        self.warehouse = warehouse
+        self.file_path = file_path
+        self.autocommit = autocommit
+        self.private_key = private_key
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs("file_path", "autocommit")
+    def run(self, file_path: str = None, autocommit: bool = None):
```

Comment:
This won't allow users to pass the password as a Secret. All of the options should be available at runtime.